src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -112,8 +112,12 @@ def create(cmd, vm_name, resource_group_name, repair_password=None, repair_usern
             # Copy OS Disk
             logger.info('Copying OS disk of source VM...')
             copy_disk_id = _call_az_command(copy_disk_command).strip('\n')
-            # Add copied OS Disk to VM creat command so that the VM is created with the disk attached
-            create_repair_vm_command += ' --attach-data-disks {id}'.format(id=copy_disk_id)
+            # For Linux the disk gets not attached at VM creation time. To prevent an incorrect boot state it is required to attach the disk after the VM got created.
+            if is_linux:
+                pass
+            else:
+                # Add copied OS Disk to VM creat command so that the VM is created with the disk attached
+                create_repair_vm_command += ' --attach-data-disks {id}'.format(id=copy_disk_id)
```

Comment:
```suggestion
            if not is_linux:
                # Add copied OS Disk to VM creat command so that the VM is created with the disk attached
                create_repair_vm_command += ' --attach-data-disks {id}'.format(id=copy_disk_id)
```